external-api/src/middleware/check-api-key/index.ts

```diff
@@ -0,0 +1,40 @@
+import dotenv from 'dotenv';
+import { Request, Response, NextFunction } from 'express';
+
+dotenv.config();
+
+const { API_KEY } = process.env;
+
+/**
+ * Check that the x-api-key header is valid
+ * @param {Object} req Request object
+ * @param {Object} res Response object
+ * @param {String} next Callback function name
+ */
+export const checkApiKey = (req: Request, res: Response, next: NextFunction) => {
+  const {
+    headers: { 'x-api-key': xApiKey },
+  } = req;
+
+  /**
+   * x-api-key is not provided.
+   * Reject the request.
+   */
+  if (!xApiKey) {
+    return res.status(401).send();
```

Comment:
Can we change to `400` please.

Reply:
Can we return a string message please with text `Bad request` please.

Reply:
It was originally `400` as per conversations, but was asked to change to `401`: https://github.com/UK-Export-Finance/dtfs2/pull/1768#discussion_r1244095254

Have added messaging.